src/Functions/DateTimeTransforms.h

```diff
@@ -624,6 +667,16 @@ struct TimeSlotImpl
         return dateIsNotSupported(name);
     }
 
+    static inline DecimalUtils::DecimalComponents<DateTime64> execute_extended_result(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl & time_zone)
+    {
+       return {time_zone.toStartOfMinuteInterval(t.whole, 30), 0};
+    }
+
+    static inline Int64 execute_extended_result(Int32, const DateLUTImpl &)
+    {
+        return date32IsNotSupported(name);
```

Comment:
Looks a bit weird that the `*IsNotSupported()` methods return an integer. They ideally return `void` and we could just call `date32IsNotSupported()` here. But perhaps it is because we return Int64 in l. 675 and the compiler gets confused otherwise.

Reply:
[[noreturn]] probably helps